lib/fizzy/instantiate.hpp

```diff
@@ -117,11 +117,47 @@ struct ImportedFunction
     std::function<ExecutionResult(Instance&, const Value*, int depth)> function;
 };
 
-// Create vector of ExternalFunctions ready to be passed to instantiate.
-// imported_functions may be in any order,
-// but must contain functions for all of the imported function names defined in the module.
-std::vector<ExternalFunction> resolve_imported_functions(
-    const Module& module, std::vector<ImportedFunction> imported_functions);
+struct ImportedTable
+{
+    std::string module;
+    std::string name;
+    table_elements* table = nullptr;
+    uint32_t min = 0;
+    std::optional<uint32_t> max;
+};
+
+struct ImportedMemory
+{
+    std::string module;
+    std::string name;
+    bytes* data = nullptr;
+    uint32_t min = 0;
+    std::optional<uint32_t> max;
+};
+
+struct ImportedGlobal
+{
+    std::string module;
+    std::string name;
+    Value* value = nullptr;
+    ValType type = ValType::i32;
+    bool is_mutable = false;
+};
+
+struct Externals
+{
+    std::vector<ExternalFunction> functions;
+    std::vector<ExternalTable> tables;
+    std::vector<ExternalMemory> memories;
+    std::vector<ExternalGlobal> globals;
+};
+
+// Create vectors of External* ready to be passed to instantiate.
+// imported_* may be in any order,
+// but must contain imports for all of the import names defined in the module.
+Externals resolve_imports(const Module& module, std::vector<ImportedFunction> imported_functions,
```

Comment:
The `std::span` should be used as the parameters type, but I'm not sure how good our implementation is.

Reply:
I tried it, and `span` doesn't look suitable here, because it's supposed to be created over existing container (it doesn't have an `initializer_list` constructor, because `initializer_list` is not a container, but just a temporary), so the `vector` would need to be explicitly created on the caller side. Whereas now it's possible to pass just an initializer list.

Now:
```
        auto imports = fizzy::resolve_imported_functions(
            *module, {
                         {"env", "adler32", {fizzy::ValType::i32, fizzy::ValType::i32},
                             fizzy::ValType::i32, env_adler32},
                     });
```

With `span`:
```
        auto imports = fizzy::resolve_imported_functions(
            *module, std::vector<ImportedFunction>{
                         {"env", "adler32", {fizzy::ValType::i32, fizzy::ValType::i32},
                             fizzy::ValType::i32, env_adler32},
                     });
```

Reply:
I think I'll leave it as vector const reference.